text/0779-first-class-component-templates.md

```diff
@@ -0,0 +1,1401 @@
+
+Even, this is a real gap, which we could address in a future RFC. Notably, it is *not specific to this proposal*, but applies to *all* proposals built on the current primitives.
+
+
+### Interop
+
+Since all existing components already work using the same low-level primitives the new system uses, strict-mode components using `<template>` can import and invoke components authored in Ember’s loose mode format. Similarly, since loose mode components are resolved to a component which is the default export of a module in the correct/conventional location on disk, components authored in strict mode with `<template>` and exported as the default export in that conventional location will be resolveable by loose mode components as well.
+
+However, other components exported as *named* exports will not be available in loose mode. This is a temporary incoherence which will be resolved as the ecosystem migrates to strict mode. As a workaround, users may choose to create reexport-only modules to allow loose-mode access. For example, given a module `app/components/named-only.js` with two named export components:
+
+```js
+export const Greet = <template>
+  <p>Hello, {{@name}}!</p>
+</template>
+
+export const Farewell = <template>
+  <p>Hello, {{@name}}!</p>
+</template>
+```
+
+A user wishing to make these available to loose mode could introduce two new modules:
+
+- `app/components/named-only/greet.js`:
+
+    ```js
+    export { Greet as default } from '../named-only';
+    ```
+
+- `app/components/named-only/farewell.js`:
+
+    ```js
+    export { Farewell as default } from '../named-only';
+    ```
+
+Then a loose-mode component could invoke them like `<NamedOnly::Greet @name="Chris">` and `<NamedOnly::Farewell @name="Krycho" />`. Note that this pattern is not at all *necessary* for migration, but may be useful.
+
+
+### The “prelude”
+
+While all values used in templates must be explicitly in scope, Ember[^glimmer-prelude] will provide some via a “prelude”.[^prelude] These are always in scope and do not need to be imported.
+
+The helpers in the prelude are:
+
+- `action`
+- `array`
+- `component`
+- `concat`
+- `debugger`
+- `each`
+- `each-in`
+- `fn`
+- `get`
+- `has-block`
+- `has-block-params`
+- `hash`
+- `if`
+- `in-element`
+- `let`
+- `log`
+- `mount`
+- `mut`
+- `on`
+- `outlet`
+- `query-params`
+- `unbound`[^unbound-deprecated]
+- `unless`
+- `yield`
+
+The modifiers in the prelude are:
+
+- `on`
+
+The components in the prelude are:
+
+- `LinkTo`
+- `Input`
+- `Textarea`
+
+[^glimmer-prelude]: Glimmer.js may provide its own prelude. While long-term the two should likely align, this RFC simply takes the _status quo_ as a given
+
+[^prelude]: “Prelude” is the conventional name for this functionality in programming language design.
+
+[^unbound-deprecated]: `unbound` is deprecated, but since it will not be removed till Ember 5.0.0, it should appear in the prelude nonetheless.
```

Comment:
Hi @chriskrycho! AFAIK, `unbound` is not yet deprecated. There's [an open RFC](https://github.com/emberjs/rfcs/pull/727) for it though.
Would be nice if we can push that one forward as well some time!

Reply:
Ah! I was getting ahead of myself it seems. I just really _want_ it gone. 😆 I'll update, and thank you!